realm-jni/src/io_realm_internal_table.cpp

```diff
@@ -641,6 +618,12 @@ JNIEXPORT void JNICALL Java_io_realm_internal_Table_nativeSetString(
     if (!TBL_AND_INDEX_AND_TYPE_VALID(env, TBL(nativeTablePtr), columnIndex, rowIndex, type_String))
         return;
     try {
+        if (value == NULL) {
```

Comment:
This is a good candidate for a function or macro to simplify the code as this snippet is used many places.